File: 0704-binary-search/0704-binary-search.java
class Solution {
    public int search(int[] nums, int target) {
         int low=0,high=nums.length-1;
        while (low<=high){
            int pivot=low+(high-low)/2;
            if (nums[pivot]==target)return pivot;
            else if (nums[pivot]<target)low=pivot+1;
            else high=pivot-1;
        }
        return -1;
    }
}

File: 0704-binary-search/0704-binary-search.py
class Solution:
    def search(self, nums: List[int], target: int) -> int: 
        L,R= 0,len(nums)-1 
        while L<=R:
            M=(L+R)//2
            if nums[M]==target:
                return M
            if nums[M]>target:
                R=M-1
            else:
                L=M+1
        return -1